source/extensions/outlier_detection_monitors/common/monitor_base_impl.h

```diff
@@ -54,39 +54,62 @@ class LocalOriginEventsBucket : public TypedErrorsBucket<Upstream::Outlier::Loca
   bool match(const ExtResult&) const override;
 };
 
+/*
+    Base config for all types of monitors.
+*/
+class ExtMonitorConfig {
+public:
+  ExtMonitorConfig(const std::string& name, uint32_t enforce) : name_(name), enforce_(enforce) {}
+  void processBucketsConfig(
+      const envoy::extensions::outlier_detection_monitors::common::v3::ErrorBuckets& config);
+  void addErrorBucket(ErrorsBucketPtr&& bucket) { buckets_.push_back(std::move(bucket)); }
```

Comment:
Seems both these two methods could be merged to the constructor of `ExtMonitorConfig`

Reply:
I had to rework the API (protos)  to do that. See the comment in the main thread.